gateway-util-urltemplate/src/test/java/org/apache/knox/gateway/util/urltemplate/MatcherTest.java

```diff
@@ -796,7 +796,7 @@ public void testMultipleDoubleStarPathMatching() throws URISyntaxException {
     Template template;
     Template input;
     Matcher<String> stringMatcher;
-    Matcher<?>.Match match;
```

Comment:
Is this change necessary?

Reply:
Like I indicated in the description without this change I was not able to run tests locally. I'm not sure how it was working before in Travis CI..

Reply:
Yea do you by chance have the Java level not set to 7+ or something where it doesn't like the generics?

Reply:
```
$ java -version
java version "1.8.0_151"
Java(TM) SE Runtime Environment (build 1.8.0_151-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.151-b12, mixed mode)
```
I use the same version in my IDE too...

The same issue like described here: https://stackoverflow.com/questions/29621138/why-is-assertthatmap1-sameinstancemap2-not-allowed